core/endpoint.go

```diff
@@ -364,6 +364,18 @@ func (endpoint *Endpoint) call(result *Result) {
 	}
 }
 
+/*
+This function is using to close HTTP connection between watchdog and endpoints
+So we can avoid dangling socket file descriptors
+If we find similar issue to this in the future, we can add the disconnection method in this function
+Reference: https://github.com/TwiN/gatus/issues/536
+*/
+func (endpoint *Endpoint) Close() {
```

Comment:
```suggestion
// Close closes HTTP connections between watchdog and endpoints to avoid dangling socket file descriptors
// on configuration reload.
// More context on https://github.com/TwiN/gatus/issues/536
func (endpoint *Endpoint) Close() {
```